src/components/toast/toast.spec.ts

```diff
@@ -0,0 +1,162 @@
+        });
+
+        describe(`When the Toast is created`, () => {
+            it(`Should automatically appear`, () => {
+                expect(((wrapper.vm as unknown) as PortalMixin).propOpen).toBeTruthy();
+                expect(((wrapper.vm as unknown) as Portal).portalCreated).toBeTruthy();
+                expect(((wrapper.vm as unknown) as Portal).portalMounted).toBeTruthy();
+            });
+
+            it(`Should be in Confirmation state`, () => {
+                expect(wrapper.vm.state).toEqual(MMessageState.Confirmation);
+            });
+
+            it(`Should be in the bottom-right position`, () => {
+                expect(wrapper.vm.position).toEqual(MToastPosition.BottomRight);
+            });
+
+            it(`Should have a timeout set to 0`, () => {
+                expect(wrapper.vm.timeout).toEqual(0);
+            });
+
+            it(`Should have an icon`, () => {
+                expect(wrapper.vm.icon).toBeTruthy();
+            });
+
+            it(`Should have an offset set to 0 as a string`, () => {
+                expect(wrapper.vm.offset).toEqual('0');
+            });
+        });
+
+        describe(`When the close button is clicked`, () => {
+            it(`Should emit a close event`, () => {
+                wrapper.find('.m-toast__close-button').trigger('click');
+
+                expect(wrapper.emitted('close')).toBeTruthy();
+            });
+        });
+    });
+
+    describe(`Given that a custom action prop have been passed`, () => {
+        beforeEach(() => {
+            actionLabel = ACTION_LABEL;
+            initializeShallowWrapper();
+            jest.runOnlyPendingTimers(); // wait for component to be instancialized
+        });
+
+        describe(`When the Toast is created`, () => {
+            it(`Should have an action label`, () => {
+                expect(wrapper.vm.actionLabel).toBe(ACTION_LABEL);
+            });
+
+            it(`Should show a button with the label on it`, () => {
+                const label: string = wrapper.find('.m-toast__actions').html();
+
+                expect(label).toContain(ACTION_LABEL);
+            });
+        });
+
+        describe(`When the action button is clicked`, () => {
+            it(`Should emit a action event`, () => {
+                wrapper.find('.m-toast__actions').find('m-link').trigger('click');
+
+                expect(wrapper.emitted('action-button')).toBeTruthy();
+            });
+        });
+    });
+
+    describe(`Given that a timeout prop have been passed`, () => {
+        describe(`When the Toast is created`, () => {
+            it(`Should appear and then disappear`, () => {
+                timeout = 5000;
+                initializeShallowWrapper();
+                jest.runOnlyPendingTimers(); // wait for component to be instancialized
+
+                expect(((wrapper.vm as unknown) as PortalMixin).propOpen).toBeTruthy();
```

Comment:
`wrapper.vm as PortalMixin` ne fonctionne pas?

Reply:
Non, mon IDE crie au meurtre 🔪 

> Conversion of type 'MToast' to type 'PortalMixin' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
>   Property 'propOpen' is missing in type 'MToast'. [2352]